packages/eslint-plugin/docs/rules/consistent-type-assertions.mdx

```diff
@@ -42,8 +42,6 @@ Some codebases like to go for an extra level of type safety, and ban assertions
 
 ### `objectLiteralTypeAssertions`
 
-Whether to always prefer type declarations for object literals used as variable initializers, rather than type assertions.
```

Comment:
Hmm. I think this implicit injection makes the rule docs source hard to understand. If we do inject like this, can it be something explicitly reflected in the source, so that the source reads more like the following? 

```mdx
### `objectLiteralTypeAssertions`

<RuleOptionDescription for='consistent-type-assertions' option='objectLiteralTypeAssertions'/>

For example, this would prefer `const x: T = { ... };` to `const x = { ... } as T;` (or similar with angle brackets).

(etc)
```

I don't have a lot of familiarity with best practices here, but I'm worried that the change as-is really hampers understandability of the doc page source especially for community contributors.

Reply:
Hmm, I can see what you're saying... but that's so much manual work for something we _know_ for certain on each rule's option. 😬

Even if we pare it down to `<RuleOptionDescription option="objectLiteralTypeAssertions" />` using current file info (which we can), we'd then also have to write a unit test or lint rule to make sure they don't fall out of sync.

The rule docs are already pretty automated. My leaning is towards continuing that.

![Arbiter and Master Chief back to back in Halo 3, with Gravemind tentacles behind them. Arbiter is speaking: "we trade one villain for another"](https://github.com/user-attachments/assets/749b5be4-8a9f-4868-a106-c7d3bb7a6284)

Reply:
Chatted a bit with @Josh-Cena and @kirkwaiblinger - I'll switch to something more explicit. Both are in favor of making it more clear where stuff will be injected.